### PR TITLE
13 mvp son list view

### PR DESCRIPTION
**Pull request by philwilliammee**
#13 
adds kyslik/column-sortable package
https://github.com/Kyslik/column-sortable

may need to run
`php artisan vendor:publish --provider="Kyslik\ColumnSortable\ColumnSortableServiceProvider" --tag="config"`

 ✔️ List view includes timestamp, title, college/unit, submitter
 ✔️ List view can be sorted by these fields
 ✔️ List view is paginated to 50 items (but test with small number)
 ✔️ Detail view has all fields

### CODE REVIEW

**Comment by philwilliammee** [comment]
oops. I forgot to checkout branch #14 sorry. commits 

95952f5 & 
eaa1964 
close out issue #14

✔️ As an admin user, I can update a SON to indicate current status

Requirements

✔️ Admin user can set a status field for each SON to indicate current state of review
✔️  List view shows current **status** and can be filtered by it

File: app/Http/Controllers/SonController.php [hunk 1]
```diff
@@ -3,18 +3,42 @@
...
     /**
      * Display a listing of the resource.
      *
      * @return \Illuminate\Http\Response
      */
-    public function index()
+    public function index(Son $son)
     {
-        $sons = Son::orderBy('date', 'desc')->simplePaginate(50);
+        $sons = $son->sortable()->paginate(50);
```
Comment (woodseowl) [inline app/Http/Controllers/SonController.php:41]: We lost the orderBy. How do we get it back? date descending is good.

File: app/Son.php [hunk 2]
```diff
@@ -3,8 +3,12 @@
...
 
 use Illuminate\Database\Eloquent\Model;
+use Kyslik\ColumnSortable\Sortable;
 
 class Son extends Model
 {
+    use Sortable;
+
     protected $guarded = ['id'];
+    public $sortable = ['title', 'user_netid', 'created_at', 'college_name'];
```
Comment (woodseowl) [inline app/Son.php:13]: I tried having 'created_at' be first to see if this fixed the sort order, but it also needs 'desc', which I don't see how to identify.

File: resources/views/home.blade.php [hunk 3]
```diff
@@ -1,51 +1,68 @@
...
                 <tr>
-                    <th></th>
-                    <th scope="col">Submitted date</th>
-                    <th scope="col">Title</th>
-                    <th scope="col">Submitted by</th>
+                    <th scope="col"> @sortablelink('title', 'Title') </th>
+                    <th scope="col"> @sortablelink('user_netid', 'Submitter')</th>
+                    <th scope="col"> @sortablelink('college_name', 'College/Unit')</th>
+                    <th scope="col"> @sortablelink('status', 'Status')</th>
+                    <th scope="col"> @sortablelink('created_at', 'Ttimestamp')</th>
```
Comment (woodseowl) [inline resources/views/home.blade.php:35]: typo

```suggestion
                    <th scope="col"> @sortablelink('created_at', 'Timestamp')</th>
```

File: resources/views/home.blade.php [hunk 4]
```diff
@@ -1,51 +1,68 @@
...
                     <tr>
-                        <th scope="row">{{ $son->id }}</th>
-                        <td>{{ $son->created_at->format('Y-F-j')}}</td>
-                        <td>
-                            <a href="{{route('sons.show', $son->id)}}">{{$son->title}}</a> 
+                        <td><a href="{{route('sons.show', $son->id)}}">{{$son->title}}</a></td>
                         <td>{{ $son->user_netid }}</td>
+                        <td>{{ $son->college_name }}</td>
+                        <td>{{ $son->status}}</td>
+                        <td>{{ $son->created_at->format('Y-m-j')}}</td>
```
Comment (woodseowl) [inline resources/views/home.blade.php:46]: Dates are coming out like "2020-04-9", which is weird. Just going with "2020-04-09".
```suggestion
                        <td>{{ $son->created_at->format('Y-m-d')}}</td>
```

File: app/Helpers/ActiveDirectory.php [hunk 5]
```diff
@@ -0,0 +1,78 @@
+<?php
+
```
[the remark this answers is not in the document]
Comment (philwilliammee) [reply app/Helpers/ActiveDirectory.php:2]: yeah lets remove it

File: app/Http/Controllers/SonController.php [hunk 6]
```diff
@@ -3,18 +3,42 @@
...
 
 class SonController extends Controller
 {
+    /**
+     * @inheritdoc
+     */
+    public function __construct()
+    {
+        $this->middleware('auth');
+        $this->bc1 = (object)[
```
[the remark this answers is not in the document]
Comment (philwilliammee) [reply app/Http/Controllers/SonController.php:23]: yeah thats a bead one, it means bread_crumb it affects breadcrumbs.blade.php I'll put it in as an issue

[the remark this answers is not in the document]
Comment (philwilliammee) [reply app/Http/Controllers/SonController.php:23]: #30 issue put in

File: app/Http/Controllers/SonController.php (hunk 1 above)
Comment (philwilliammee) [reply app/Http/Controllers/SonController.php:44]: pass it ot sortable as an argument.
```php
$sons = $son->sortable(['title' => 'desc'])->paginate(50);
```

File: app/Helpers/ActiveDirectory.php (hunk 5 above)
[the remark this answers is not in the document]
Comment (philwilliammee) [reply app/Helpers/ActiveDirectory.php:2]: #31 ticket submitted

File: app/Http/Controllers/SonController.php (hunk 1 above)
Comment (philwilliammee) [reply app/Http/Controllers/SonController.php:44]: issue #32

File: app/Son.php (hunk 2 above)
Comment (philwilliammee) [reply app/Son.php:13]: addressed in #32

File: resources/views/home.blade.php (hunk 4 above)
Comment (philwilliammee) [reply resources/views/home.blade.php:46]: I was trying to keep column width as narrow as possible

**Comment by philwilliammee** [comment]
I think I approved all change requests, something is blocking though. Still trying to work through the nuances of this process.

**Comment by woodseowl** [comment]
Thanks, Phil! The issues added are things I can address. I'll approve as is and work to get those issues next week.

**Pull request merged**